fe/fe-core/src/main/java/com/starrocks/sql/PlannerProfile.java

```diff
@@ -0,0 +1,133 @@
+// This file is made available under Elastic License 2.0.
+
+package com.starrocks.sql;
+
+import com.google.common.base.Preconditions;
+import com.starrocks.common.util.RuntimeProfile;
+import com.starrocks.qe.ConnectContext;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class PlannerProfile {
+    private ConnectContext ctx;
+
+    public static class ScopedTimer implements AutoCloseable {
+        private long startTime = 0;
+        private volatile long currentThreadId = 0;
+        private long totalTime = 0;
+        private int totalCount = 0;
+        // possible to record p99?
+
+        public void start() {
+            Preconditions.checkState(currentThreadId == 0);
+            currentThreadId = Thread.currentThread().getId();
+            startTime = System.currentTimeMillis();
+        }
+
+        public void close() {
+            Preconditions.checkState(currentThreadId == Thread.currentThread().getId());
+            currentThreadId = 0;
+            totalTime += (System.currentTimeMillis() - startTime);
+            totalCount += 1;
+        }
+
+        public long getTotalTime() {
+            return totalTime;
+        }
+
+        public int getTotalCount() {
+            return totalCount;
+        }
+    }
+
+    private final HashMap<String, ScopedTimer> timers;
+
+    public PlannerProfile() {
+        timers = new HashMap<>();
+    }
+
+    public void init(ConnectContext ctx) {
+        this.ctx = ctx;
+    }
+
+    private synchronized ScopedTimer getOrCreateScopedTimer(String name) {
+        if (timers.containsKey(name)) {
+            return timers.get(name);
+        }
+        ScopedTimer t = new ScopedTimer();
+        timers.put(name, t);
+        return t;
+    }
+
+    private static final PlannerProfile DEFAULT_INSTANCE = new PlannerProfile();
+
+    public static ScopedTimer getScopedTimer(String name) {
+        // to avoid null.
+        PlannerProfile p = DEFAULT_INSTANCE;
```

Comment:
I perfer to assert null

Reply:
No. I guess in some UT, `ConnectContext` is null, and we don't have `PlannerProfile`, and `getScopedTimer` will throw NPE.

Reply:
Suppose in UT, we are running following code

```
try (ScopedTimer _ =  GetScopedTimer())  {
}
```

If we return NULL(or assert not null), we will get NPE.